File: src/main/resources/static/app/classes/Follow.ts
/**
 * Class that represent the information related to MiausicBox follows
 * @class Follow
 */
import { User } from './User';

export class Follow {

  /* Attributes */
  private _id:number;
  private _emisor:User;
  private _receptor:User;

  /* Constructor */
  constructor(id:number, emisor:User, receptor:User) {
    this._id = id;
    this._emisor = emisor;
    this._receptor = receptor;
  }

  /* Getters & Setters */
  get id():number {
    return this._id;
  }

  get emisor():User {
    return this._emisor;
  }

  get receptor():User {
    return this._receptor;
  }

  set id(id:number) {
    this._id = id;
  }

  set emisor(emisor:User) {
    this._emisor = emisor;
  }

  set receptor(receptor:User) {
    this._receptor = receptor;
  }

  /** Return if an Follow is equal to other
   *  @method equals
   *  @param {object} object
   */
  equals(object:any) {
    if (!(object instanceof Follow)) {
      return false;
    } else {
      var follow:Follow = object;
      return (follow.emisor.equals(this.emisor) &&
              follow.receptor.equals(this.receptor));
    }
  }

}

File: src/main/java/giraffe/miausicbox/model/Follow.java
package giraffe.miausicbox.model;

import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Follow {

	public interface BasicAtt {}
	
	// Attributes
	@JsonView(BasicAtt.class)
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	
	@JsonView(BasicAtt.class)
	@OneToOne
	private User emisor;
	
	@JsonView(BasicAtt.class)
	@OneToOne
	private User receptor;
	
	// Constructor
	public Follow() {}
	
	public Follow(
			User emisor,
			User receptor
			) {
		this.emisor = emisor;
		this.receptor = receptor;
	}

	// Getters & Setters
	public long getId() {
		return id;
	}

	public User getEmisor() {
		return emisor;
	}

	public void setEmisor(User emisor) {
		this.emisor = emisor;
	}

	public User getReceptor() {
		return receptor;
	}

	public void setReceptor(User receptor) {
		this.receptor = receptor;
	}
	
	// ToString
	@Override
	public String toString() {
		return (this.getEmisor().getUserName() + " - " + this.getReceptor().getUserName());
	}
	
	// Equals
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final Follow other = (Follow) obj;
		if (!Objects.equals(this.getId(), other.getId())) {
			return false;
		}
		return (Objects.equals(this.getEmisor(), other.getEmisor())
				&& Objects.equals(this.getReceptor(), other.getReceptor()));
	}
	
}
